site/src/hooks/usePaginatedQuery.test.ts

```diff
@@ -0,0 +1,407 @@
+import { renderHookWithAuth } from "testHelpers/renderHelpers";
+import { waitFor } from "@testing-library/react";
+
+import {
+  type PaginatedData,
+  type UsePaginatedQueryOptions,
+  usePaginatedQuery,
+} from "./usePaginatedQuery";
+
+beforeAll(() => {
+  jest.useFakeTimers();
+});
+
+afterAll(() => {
+  jest.useRealTimers();
+  jest.clearAllMocks();
+});
+
+function render<
+  TQueryFnData extends PaginatedData = PaginatedData,
+  TQueryPayload = never,
+>(
+  options: UsePaginatedQueryOptions<TQueryFnData, TQueryPayload>,
+  route?: `/?page=${string}`,
+) {
+  return renderHookWithAuth(({ options }) => usePaginatedQuery(options), {
+    route,
+    path: "/",
+    initialProps: { options },
+  });
+}
+
+/**
+ * There are a lot of test cases in this file. Scoping mocking to inner describe
+ * function calls to limit the cognitive load of maintaining all this stuff
+ */
+describe(`${usePaginatedQuery.name} - Overall functionality`, () => {
```

Comment:
nit: I feel like "overall functionality" is to be assumed if you're not mentioning something more specific